geode-core/src/main/java/org/apache/geode/internal/JarDeployer.java

```diff
@@ -386,6 +402,98 @@ public DeployedJar findLatestValidDeployedJarFromDisk(String unversionedJarName)
     return deployedJars;
   }
 
+  private String getJdbcDriverName(DeployedJar jar) {
+    File jarFile = jar.getFile();
+    try {
+      FileInputStream fileInputStream = new FileInputStream(jarFile.getAbsolutePath());
+      BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream);
+      ZipInputStream zipInputStream = new ZipInputStream(bufferedInputStream);
+      ZipEntry zipEntry;
+      while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+        // JDBC 4.0 Drivers must include the file META-INF/services/java.sql.Driver. This file
+        // contains the name of the JDBC drivers implementation of java.sql.Driver
+        // See https://docs.oracle.com/javase/8/docs/api/java/sql/DriverManager.html
+        if (!zipEntry.getName().equals("META-INF/services/java.sql.Driver")) {
+          continue;
+        }
+        int size = (int) zipEntry.getSize();
+        if (size == -1) {
+          logger.warn("Invalid zip entry found for META-INF/services/java.sql.Driver within jar "
+              + jarFile.getName());
+          return null;
+        }
+        byte[] bytes = new byte[size];
+        int offset = 0;
+        int chunk;
+        while ((size - offset) > 0) {
+          chunk = zipInputStream.read(bytes, offset, size - offset);
+          if (chunk == -1) {
+            break;
+          }
+          offset += chunk;
+        }
+        return new String(bytes);
+      }
+      return null;
+    } catch (IOException ex) {
+      return null;
+    }
+  }
+
+  // DriverManager only uses a driver loaded by system ClassLoader
+  class DriverWrapper implements Driver {
+
+    private Driver jdbcDriver;
+
+    DriverWrapper(Driver jdbcDriver) {
+      this.jdbcDriver = jdbcDriver;
+    }
+
+    public Connection connect(String url, java.util.Properties info)
+        throws SQLException {
+      return this.jdbcDriver.connect(url, info);
+    }
+
+    public boolean acceptsURL(String url) throws SQLException {
+      return this.jdbcDriver.acceptsURL(url);
+    }
+
+    public DriverPropertyInfo[] getPropertyInfo(String url, java.util.Properties info)
+        throws SQLException {
+      return this.getPropertyInfo(url, info);
+    }
+
+    public int getMajorVersion() {
+      return this.getMajorVersion();
+    }
+
+    public int getMinorVersion() {
+      return this.jdbcDriver.getMinorVersion();
+    }
+
+    public boolean jdbcCompliant() {
+      return this.jdbcDriver.jdbcCompliant();
+    }
+
+    public java.util.logging.Logger getParentLogger() throws SQLFeatureNotSupportedException {
+      return this.jdbcDriver.getParentLogger();
+    }
+  }
+
+
+  private void addToSystemClasspathAndRegisterDriver(DeployedJar jar,
+      String driverClassName) {
+    File jarFile = jar.getFile();
+    try {
+      URLClassLoader urlClassLoader = new URLClassLoader(new URL[] {jarFile.toURI().toURL()});
```

Comment:
Since the class has been deployed, you should instead do `ClassPathLoader.getLatest().forName(clazz)` instead of creating a separate classloader.

Reply:
Done